src/master-playlist-controller.js

```diff
@@ -623,10 +623,11 @@ export class MasterPlaylistController extends videojs.EventTarget {
 
     if (this.mediaTypes_.AUDIO.activePlaylistLoader) {
       // if the audio playlist loader exists, then alternate audio is active
-      if (this.mainSegmentLoader_.startingMedia_ &&
+      if (!this.mainSegmentLoader_.startingMedia_ ||
```

Comment:
What happens in the event that we only have alternate audio, it hits the end of its playlist, and `mainSegmentLoader_.startingMedia_` is not yet set?

Reply:
I do not believe this is a scenario we can get into. `STREAM-INF`s are REQUIRED to have a URI line by the spec. So there will always be a main stream that the main segment loader is loading from. In addition to that, until the way audio buffer ownership is overhauled, there isn't actually any mechanism to stop the main segment loader in an audio only stream with alternate audio activated. So even if audio only and alternate audio activated, the main segment loader will continue to request its audio segments, and then the VirtualSourceBuffer just drops them on the floor because it doesn't own the audio buffer. (Yes this is wasted bandwidth, even more reason to change the way we handle audio)